pkg/cluster/denyassignment.go

```diff
@@ -0,0 +1,49 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/deployment"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (m *manager) createOrUpdateDenyAssignment(ctx context.Context) error {
+	if m.env.DeploymentMode() != deployment.Production {
+		// only need this upgrade in production, where there are DenyAssignments
+		return nil
+	}
+
+	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+
+	clusterSPObjectID, err := m.aad.GetServicePrincipalID(ctx)
+	if err != nil {
+		return err
+	}
+
+	denyAssignments, err := m.denyAssignments.ListForResourceGroup(ctx, resourceGroup, "")
+	if err != nil {
+		return err
+	}
+
+	for _, assignment := range denyAssignments {
+		if assignment.DenyAssignmentProperties.ExcludePrincipals != nil {
+			for _, ps := range *assignment.DenyAssignmentProperties.ExcludePrincipals {
+				if *ps.ID == clusterSPObjectID {
+					return nil
+				}
+			}
+		}
+	}
+
+	t := &arm.Template{
+		Schema:         "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+		ContentVersion: "1.0.0.0",
+		Resources:      []*arm.Resource{m.denyAssignment(clusterSPObjectID)},
+	}
```

Comment:
@jim-minter do you know if denyAssignment can be updated OR it need to be re-created?
We already did updates for Actions but in this case SPID would be changing. And this field might be non-updatable.
Any ideas before we start testing in prodution?